tsweb/server.go

```diff
@@ -0,0 +1,345 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tsweb
+
+import (
+	"context"
+	"crypto/tls"
+	"expvar"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/crypto/acme/autocert"
+	"tailscale.com/metrics"
+	"tailscale.com/types/logger"
+)
+
+const defaultTimeout = 10 * time.Second
+
+// ServerConfig is the initial configuration of a Server, for
+// consumption by NewServer.
+type ServerConfig struct {
+	// Name is a human-readable, alphanumeric name for the HTTP
```

Comment:
give an example string literal in the comment (such as "foo") or (such as "foo.example.com") or (such as "Brad's Nice server")

Notably to clarify whether it's a hostname (with dots) or some bare name or some descriptive name.

Alphanumeric sounds like just "foo"?

Reply:
Added words. Basically has to be valid as part of an email address, and as a directory name.